### ISSUE TITLE
When using "transformers.WarmUp" with tensorflow 2.0.0, warming up restart in each epoch!

### ISSUE THREAD

**Issue by Wchenguang** [huggingface/transformers#5765]
When using "transformers.WarmUp" with tensorflow 2.0.0, warming up restart in each epoch!
That is because, in keras "Callbacks.on_batch_begin(self, batch, logs)", batch start from zero in each epoch when using "fit" method.

**Comment by jplu** [comment]
Hello!

Can you give a code that better explain your issue and a way to reproduce it please? Thanks :)

**Comment by Wchenguang** [comment]
Sorry! I check again and it works well!
`
import tensorflow as tf
from transformers import WarmUp
import numpy as np
inputs = tf.keras.Input(shape = (20))
outputs = tf.keras.layers.Dense(2)(inputs)
model = tf.keras.models.Model(inputs = inputs,
                             outputs = outputs)
x = []
lr = []
def decay(step):
    return 0.1

class metrics(tf.keras.callbacks.Callback):
    def __init__(self):
        super().__init__()
    def on_batch_begin(self, batch, logs: dict):
        x.append(batch)
        lr.append(self.model.optimizer._decayed_lr('float').numpy())
        

schedule = WarmUp(0.1, decay_schedule_fn=decay, warmup_steps=5)
optimizer = tf.keras.optimizers.Adam(learning_rate=schedule)
model.compile(optimizer=optimizer,
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
model.fit(np.random.rand(10, 20),
         np.random.randint(0, 2, 10),
         batch_size = 1,
         epochs = 2,
         callbacks = [metrics()])
print(x, lr)
`
Train on 10 samples
Epoch 1/2
10/10 [==============================] - 0s 26ms/sample - loss: 6.6085 - accuracy: 0.4000
Epoch 2/2
10/10 [==============================] - 0s 2ms/sample - loss: 4.7910 - accuracy: 0.7000
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9] [0.0, 0.020000001, 0.040000003, 0.060000002, 0.080000006, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
